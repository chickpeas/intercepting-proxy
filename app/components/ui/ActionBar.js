// @flow
import React, { Component } from 'react';
import Switch from 'react-switch';
import styles from './ActionBar.scss';

// TODO:
// TEST

type Props = {
  handleClick: (boolean) => void,
  handleForwardClick: (boolean) => void,
  pendingRequest: Array,
  filter: Object
};

export default class ActionBar extends Component<Props> {
  props: Props;

  render() {
    const { filter, pendingRequest } = this.props;
    const disabled = !(filter.intercept && pendingRequest.length !== 0);

    return (
      <div className={styles.barContainer}>
        <label className={styles.labelBox} htmlFor="intercept-switch">
          <div className={styles.center}>Intercept</div>
          <Switch
            checked={filter.intercept}
            onChange={(checked) => {
            this.props.handleClick(checked);
            }
          }
          />
        </label>
        <button
          className={styles.forward}
          onClick={this.props.handleForwardClick}
          disabled={disabled}
        >
            Forward
        </button>
      </div>
    );
  }
}
