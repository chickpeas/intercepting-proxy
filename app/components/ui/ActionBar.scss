.barContainer {
  display: flex;
  flex-grow: 1;
  flex-basis: 100%;
  padding: 5px;
  justify-content: space-around;
  background-color: #fff;
  border-radius: 2px;
  margin-bottom: 5px;

  .labelBox {
    flex-direction: row;
    justify-content: center;
    align-items: center;
    display: flex;
  }
  .center {
    align-self: center;
    padding-right: 10px;
  }

  // TODO some decent theming
  .forward {
    color: rgba(0, 0, 0, 0.87);
    padding: 6px 14px;
    box-shadow: 0px 1px 5px 0px rgba(0, 0, 0, 0.2), 0px 2px 2px 0px rgba(0, 0, 0, 0.14), 0px 3px 1px -2px rgba(0, 0, 0, 0.12);
    background-color: #e0e0e0;
    border: 2px solid rgb(0, 136, 0);
    text-transform: uppercase;
    font-weight: bold;
    border-radius: 2px;
    transition-property: background-color, box-shadow;
    transition-duration: 250ms, 250ms;
    transition-timing-function: cubic-bezier(0.4, 0, 0.2, 1), cubic-bezier(0.4, 0, 0.2, 1);
    box-sizing: border-box;
  }
  .forward:hover {
    background-color: #d5d5d5;
  }
  .forward:disabled {
    background-color: #e0e0e0;
    color: rgba(0, 0, 0, 0.3);
    border: 2px solid transparent;
  }
}
