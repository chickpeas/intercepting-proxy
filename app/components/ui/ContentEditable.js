import React, { Component } from 'react';

type Props = {
  endEditing: () => void,
  children: React.Node
};

const makeContentEditable = WrappedComponent =>
  class extends Component<Props> {
    constructor(props) {
      super(props);
      this.state = {
        value: props.children,
        onEditMode: false
      };
      this.getIntoEditMode = this.getIntoEditMode.bind(this);
      this.handleEnterKey = this.handleEnterKey.bind(this);
      this.getOffEditMode = this.getOffEditMode.bind(this);
      this.changeValue = this.changeValue.bind(this);
    }
    getIntoEditMode() {
      this.setState({
        ...this.state,
        onEditMode: true
      });
    }
    getOffEditMode() {
      this.setState({
        ...this.state,
        onEditMode: false
      });
      if (this.props.endEditing) {
        this.props.endEditing(this.state.value);
      }
    }
    handleEnterKey(e) {
      if (e.keyCode === 13 || e.charCode === 13) {
        this.setState({
          ...this.state,
          onEditMode: false
        });
        if (this.props.endEditing) {
          this.props.endEditing(this.state.value);
        }
      }
    }
    changeValue(e) {
      this.setState({
        ...this.state,
        value: e.target.value
      });
    }
    render() {
      return (
        <section onClick={this.getIntoEditMode}>
          {this.state.onEditMode ? (
            <input
              type='text'
              autoFocus
              value={this.state.value}
              onChange={this.changeValue}
              onKeyPress={this.handleEnterKey}
              onBlur={this.getOffEditMode}
            />
          ) : (
            <span>{this.state.value}</span>
          )}
        </section>
      );
    }
  };

const labelize = props => <section>{props.children}</section>;

const ContentEditable = makeContentEditable(labelize);

export default ContentEditable;
