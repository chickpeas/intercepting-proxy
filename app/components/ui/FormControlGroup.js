// @flow
import React, { Component } from 'react';
import {
  FormLabel,
  FormControl,
  FormGroup,
  FormControlLabel,
  FormHelperText,
} from 'material-ui/Form';
import Checkbox from 'material-ui/Checkbox';

type Props = {
  checkboxes: {
    label?: string,
    helperText?: string,
    checks: Array
  },
  handleChange: () => void
};

export default class FormControlGroup extends Component<Props> {
  props: Props;
  state = {}

  handleChange = name => (event, checked) => {
    this.setState({ [name]: checked });
    this.props.handleChange(name, event, checked);
  };

  render() {
    const {
      checkboxes: {
        label,
        helperText,
        checks,
      }
    } = this.props;
    const checkList = checks.map(({ value, checked, label: checklabel }) =>
      (<FormControlLabel
        control={
          <Checkbox
            checked={(typeof this.state[value] === 'undefined') ? checked : this.state[value]}
            onChange={this.handleChange(value)}
            value={value}
          />
        }
        label={checklabel}
      />));

    return (
      <FormControl component="fieldset">
        <FormLabel component="legend">{label}</FormLabel>
        <FormGroup>
          {checkList}
        </FormGroup>
        <FormHelperText>{helperText}</FormHelperText>
      </FormControl>
    );
  }
}
