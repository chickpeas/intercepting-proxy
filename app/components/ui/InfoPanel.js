import React, { Component } from 'react';
import styles from './InfoPanel.scss';
import Close from './../../img/close.svg';

type Props = {
  selected: Object,
  handleClose: () => {}
};


class InfoPanel extends Component<Props> {
  props: Props;

  nestedRender(obj) {
    const keys = Object.keys(obj);
    const flatten = keys.map((key) => {
      if (typeof obj[key] === 'object') {
        const flattenInner = this.nestedRender(obj[key]);
        return (
          <span key={key} className={styles[key] ? styles[key] : key}>
            <span className={styles.label}>{key}</span>
            {flattenInner}
          </span>);
      }
      return (
        <span className={styles[key] ? styles[key] : key} key={key}>
          <span className={styles.label}>{key}</span>
          <span className={styles.value}>{obj[key]}</span>
        </span>);
    });
    return (flatten);
  }

  render() {
    const { selected, handleClose } = this.props;
    const value = this.nestedRender(selected);

    return (
      <div className={styles.infoContainer}>
        <Close className={styles.closeIcon} onClick={handleClose} />
        { value }
      </div>
    );
  }
}

export default InfoPanel;
