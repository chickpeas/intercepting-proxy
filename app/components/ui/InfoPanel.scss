.infoContainer {
  display: flex;
  flex-direction: column;
  flex-grow: 1;
  flex-shrink: 1;
  border-radius: 3px;
  border: 1px solid #ccc;
  flex-basis: 30%;
  position: relative;

  span {
    display: flex;
    flex-direction: column;
    padding-left: 20px;
  }
  .requestId {display: none;}

  .label {
    font-weight: bold;
  }
  .value {}
  .closeIcon {
    width:15px;
    height: 15px;
    position: absolute;
    top:10px;
    right:10px;
    cursor: pointer;
  }
  .cookie span{
    max-width: 200px
  }
}
