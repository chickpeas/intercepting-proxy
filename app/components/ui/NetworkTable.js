import React, { Component } from 'react';
import ReactTable from 'react-table';
import styles from './NetworkTable.scss';
// TODO:
// TEST

type Props = {
  handleClick: () => void,
  network: Array
};
const columns = [{
  Header: '#',
  accessor: 'index',
  width: 40,
  filterable: false
}, {
  Header: 'method',
  accessor: 'method',
  width: 80
}, {
  Header: 'statusCode',
  accessor: 'statusCode',
  width: 80
}, {
  Header: 'url',
  accessor: 'url'
}];

class NetworkTable extends Component<Props> {
  render() {
    const {
      network
    } = this.props;

    return (
      <ReactTable
        data={network}
        columns={columns}
        className={styles.table}
        getTrProps={(state, rowInfo) => ({
            onClick: () => {
              this.props.handleClick(rowInfo);
            }
          })}
        filterable
      />
    );
  }
}

export default NetworkTable;
