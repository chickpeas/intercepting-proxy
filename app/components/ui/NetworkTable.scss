
.ReactTable {
  border: 0;
}
.ReactTable .rt-th, .ReactTable .rt-td {
  padding: 10px 7px;
}
.ReactTable .rt-thead.-header {
  padding: 10px 7px;
}
.ReactTable .rt-thead.-filters {
  padding: 10px 7px;
}
.table {
  height: 600px;
  flex-grow: 1;
  flex-shrink: 3;
}
