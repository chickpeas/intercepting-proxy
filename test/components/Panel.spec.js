import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { InfoPanel } from '../../app/components/ui/InfoPanel';

Enzyme.configure({ adapter: new Adapter() });

function setup() {
  const props = {
    request: {
      headers: 'value'
    },
    response: {
      body: 'value'
    }
  };

  const component = shallow(<InfoPanel {...props} />);
  return {
    component,
    props
  };
}

describe('InfoPanel component', () => {
  it('should render', () => {
    const { component } = setup();
    expect(component).toMatchSnapshot();
  });
});
