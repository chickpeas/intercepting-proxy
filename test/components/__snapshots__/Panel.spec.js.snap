// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoPanel component should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InfoPanel
    selected={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <test-file-stub
          className="closeIcon"
          onClick={undefined}
        />,
        Array [],
      ],
      "className": "infoContainer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "closeIcon",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "test-file-stub",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <test-file-stub
            className="closeIcon"
            onClick={undefined}
          />,
          Array [],
        ],
        "className": "infoContainer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "closeIcon",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "test-file-stub",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
